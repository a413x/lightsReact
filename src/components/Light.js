import React, { Component } from "react";

import '../styles/light.css';

class Light extends Component{
  constructor(props){
    super(props);
  }
  render(){
    let className = 'light ';
    if(this.props.isActive) className += this.props.color + '-light';
    if(this.props.blink) className += ' blink';
    return <div className = {className} />
  }
}

export default Light;
