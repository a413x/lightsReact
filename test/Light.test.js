import React from 'react'
import { shallow } from 'enzyme'

import Light from '../src/components/Light.js';

describe('Tests for Light component',()=>{

  const props = {
    isActive: false,
    blink: false,
    color:'red',
  };

  describe('Light initial',()=>{
    const light = shallow(<Light {...props} />);
    it('renders properly',()=>{
      expect(light).toMatchSnapshot();
    });
  });

  describe('Light active state',()=>{
    const nextProps = {
      isActive: true,
      blink: false,
      color:'yellow',
    };

    const light = shallow(<Light {...nextProps} />);
    it('renders properly',()=>{
      expect(light).toMatchSnapshot();
    });
  });

  describe('Light blink state',()=>{
    const nextProps = {
      isActive: true,
      blink: true,
      color:'green',
    };

    const light = shallow(<Light {...nextProps} />);
    it('renders properly',()=>{
      expect(light).toMatchSnapshot();
    });
  });

});
