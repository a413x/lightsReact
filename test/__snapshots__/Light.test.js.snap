// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Light component Light active state renders properly 1`] = `
<div
  className="light yellow-light"
/>
`;

exports[`Tests for Light component Light blink state renders properly 1`] = `
<div
  className="light green-light blink"
/>
`;

exports[`Tests for Light component Light initial renders properly 1`] = `
<div
  className="light "
/>
`;
